full-backend-tests/src/test/java/org/graylog/plugins/views/ScriptingApiResourceIT.java

```diff
@@ -610,6 +612,31 @@ void testErrorHandling() {
                 .body("message", Matchers.containsString("Failed to obtain aggregation results"));
     }
 
+    @ContainerMatrixTest
+    void testMessages() {
+        final ValidatableResponse validatableResponse = given()
+                .spec(requestSpec)
+                .when()
+                .body("""
+                        {
+                          "fields": ["source", "facility", "level"]
+                        }
+                        """)
+                .post("/search/messages")
+                .then()
+                .log().ifStatusCodeMatches(not(200))
+                .statusCode(200);
+
+        validateSchema(validatableResponse, "field: source", "string", "source");
+        validateSchema(validatableResponse, "field: facility", "string", "facility");
+        validateSchema(validatableResponse, "field: level", "numeric", "level");
```

Comment:
You could consider replacing "string" with "ResponseEntryDataType.STRING.toString()" to avoid magic-string usage.

Reply:
IDK, I probably prefer the hardcoded value, as it holds our expectation in the test without the dependency. If we change the value of ResponseEntryDataType.STRING, our test fails and we'll know that we have changed the contract - it's not "string" anymore. That wouldn't happen if we reuse the same enum value on both ends.

Reply:
The question is what we want to test here: that the value is "string" or that the value has a string from the proper enum. It is up to you to decide, I do not have a strong opinion. :)

Reply:
I'd keep the hardcoded value. That's what our users will see and expect.  What we use internally is from my perspective not important.